### PR TITLE
Changes to delete command and the way Hustlebook finds clients

### LINKED ISSUE #114: As a user, I can choose which client to delete if there are clients with similar names
So I can accurately delete clients from my HustleBook

### LINKED ISSUE #115: Bug found when finding clients with similar names
If there is a Person named "John Doe" and a Person named "John" the command `find John Doe` will show both "John" and "John Doe".

The `Find` command should only show "John Doe" in this case as the user has already typed out the full name of the Person they are finding

### LINKED ISSUE #116: Bug found when trying to edit and delete clients with similar names
If there is a Person named "John Doe" and a Person named "John", the 'Edit` and `Delete` commands that request for "John Doe" will show both "John" and "John Doe" and ask the user to specify which Person they wish to perform an action on.

The desired result is for the `Edit` or `Delete` commands to run without additional user input as the user  has already typed out the full name of the Person they wish to perform the action on

### PR DESCRIPTION
**Pull request by kuanjr**
Currently delete command deletes the first client whose name matches the given search term. This causes problems as users currenly are not able to delete the clients that are not first in the list.

There are also bugs in the way Hustlebook finds clients. If the user gives the full name of a client, only one client should match, not others with similar names.

Example: If Hustlebook contains "David Li" and "David Tan", the command `find david tan` should only show "David Tan" but not both of them
This applies to `edit` and `delete` as well.

Let's

 - Change the way clients with names matching the search term are detected in `delete`

 - Allow users to input a number to indicate the index of the client they wish to edit
 - Add logic to allow HustleBook to execute the right command when given a number as an input by checking against 
lastCommand
 - Change the way Hustlebook finds clients by matching the full search term instead of one at a time
 - Change the way `find` works by asking users to seperate different names by commas
 
Tests for this new functionality will be added at a later date

This closes #116 , closes #115, closes #114

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/NumberParser.java [hunk 1]
```diff
@@ -37,6 +38,10 @@ public Command parse() throws ParseException {
             EditCommand newCommand = (EditCommand) lastCommand;
             newCommand.setIndex(index);
             return newCommand;
+        } else if (lastCommand instanceof DeleteCommand) {
+            DeleteCommand newCommand = (DeleteCommand) lastCommand;
+            newCommand.setIndex(index);
+            return newCommand;
```
Comment (AD-NAP) [inline src/main/java/seedu/address/logic/parser/NumberParser.java:44]: If `lastCommand` is already an instance of `DeleteCommand`, I do not think there is a need for line 42, it's possible to just `return lastCommand.setIndex(index)`. If this is true, then the entire `parse` function, can be simplified to `return lastCommand.setIndex(index)`

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -16,8 +16,9 @@
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+            + "Separate names by commas (,)"
+            + "Parameters: KEYWORD, [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " alice, bob, charlie";
```
Comment (AD-NAP) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:21]: May I check why there is a need to change this to using commas after each word?

Comment (AD-NAP) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:21]: Okay I understood by your example 
"If Hustlebook contains "David Li" and "David Tan", the command find david tan should only show "David Tan" but not both of them". 

Then can I check is it possible to execute `find David Li, David Tan, Bob, alice` and expect both David Tan and David Li to appear in the results along with any names containing bob or alice? In other words, is it possible that`KEYWORD` is not just one word.

File: src/main/java/seedu/address/logic/parser/NumberParser.java (hunk 1 above)
Comment (kuanjr) [reply src/main/java/seedu/address/logic/parser/NumberParser.java:44]: right now last command is of type Command, but setIndex is a DeleteCommand method, so I have to cast it to a Deletecommand before i am able to set index

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 2 above)
Comment (kuanjr) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:21]: yes that's how it works

File: src/main/java/seedu/address/logic/parser/NumberParser.java (hunk 1 above)
Comment (AD-NAP) [reply src/main/java/seedu/address/logic/parser/NumberParser.java:44]: got it!

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 2 above)
Comment (AD-NAP) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:21]: Nice!

**Review by AD-NAP — APPROVED** [review]
LGTM!

**Pull request merged**